File: src/main/java/no/nav/syfo/api/ressurser/azuread/VirksomhetController.java
package no.nav.syfo.api.ressurser.azuread;

import no.nav.security.oidc.api.ProtectedWithClaims;
import no.nav.syfo.api.domain.RSVirksomhet;
import no.nav.syfo.ereg.EregConsumer;
import org.springframework.web.bind.annotation.*;

import javax.inject.Inject;

import static no.nav.syfo.oidc.OIDCIssuer.AZURE;
import static org.apache.commons.lang3.text.WordUtils.capitalize;
import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;

@RestController
@RequestMapping(value = "/api/internad/virksomhet/{orgnummer}")
@ProtectedWithClaims(issuer = AZURE)
public class VirksomhetController {

    private final EregConsumer eregConsumer;

    @Inject
    public VirksomhetController(
            EregConsumer eregConsumer
    ) {
        this.eregConsumer = eregConsumer;
    }

    @GetMapping(produces = APPLICATION_JSON_VALUE)
    public RSVirksomhet getVirksomhetsnavn(@PathVariable("orgnummer") String orgnummer) {
        return new RSVirksomhet()
                .navn(capitalize(eregConsumer.virksomhetsnavn(orgnummer).toLowerCase()));
    }
}

File: src/main/kotlin/no/nav/syfo/controller/internad/virksomhet/VirksomhetController.kt
package no.nav.syfo.controller.internad.virksomhet

import no.nav.security.oidc.api.ProtectedWithClaims
import no.nav.syfo.consumer.ereg.EregConsumer
import no.nav.syfo.api.auth.OIDCIssuer.AZURE
import no.nav.syfo.domain.Virksomhetsnummer
import org.apache.commons.lang3.text.WordUtils.capitalize
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.*
import javax.inject.Inject

@RestController
@RequestMapping(value = ["/api/internad/virksomhet/{orgnummer}"])
@ProtectedWithClaims(issuer = AZURE)
class VirksomhetController @Inject constructor(
    private val eregConsumer: EregConsumer
) {
    @GetMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
    fun getVirksomhetsnavn(@PathVariable("orgnummer") orgnummer: Virksomhetsnummer): RSVirksomhet {
        return RSVirksomhet(
            navn = capitalize(eregConsumer.virksomhetsnavn(orgnummer).toLowerCase())
        )
    }
}
